### PR TITLE
Reproduce Naga's nicer error reporting for WGSL

### PR DESCRIPTION
**Pull request by almarklein**
Closes https://github.com/gfx-rs/wgpu-native/issues/104

Basically, I find myself copying WGSL code to a file and running Naga from the command line on it, because that way I get prettier errors than when pushing the (faulty) WGSL via wgpu-native's  `create_shader_module`. Would be nice if the error reports were equally good.

WIP: does not seem to help yet, except duplicating part of the error message.

### CODE REVIEW

**Comment by ElectronicRU** [comment]
shouldn't this be using `error.description()`?

**Comment by cwfitzgerald** [comment]
@ElectronicRU no, [`description`](https://doc.rust-lang.org/std/error/trait.Error.html#method.description) is deprecated in favor of display

File: wgpu-core/src/device/mod.rs [hunk 1]
```diff
@@ -1040,12 +1041,16 @@ impl<B: GfxBackend> Device<B> {
             }
             pipeline::ShaderModuleSource::Wgsl(code) => {
                 profiling::scope!("naga::wgsl::parse_str");
-                // TODO: refactor the corresponding Naga error to be owned, and then
-                // display it instead of unwrapping
                 match naga::front::wgsl::parse_str(&code) {
                     Ok(module) => (None, Some(module)),
                     Err(err) => {
                         log::error!("Failed to parse WGSL code for {:?}: {}", desc.label, err);
+                        log::error!("{}:", err);
```
Comment (kvark) [inline wgpu-core/src/device/mod.rs:1048]: err is already printed in the previous line

Comment (almarklein) [reply wgpu-core/src/device/mod.rs:1048]: Yeah I know. I just copied some lines over, without cleaning it up. At the moment this does nothing except print the error message twice :)

**Comment by cwfitzgerald** [comment]
Any progress on this?

**Comment by almarklein** [comment]
> Any progress on this?

Not from my end, I'm not sure how to proceed tbh.

**Comment by kvark** [comment]
Closed by #1760

**Pull request closed without merging**